src/main/java/frc/trigon/robot/utilities/SpeedMechanism2d.java

```diff
@@ -0,0 +1,118 @@
+            positiveBottomAngle = 145,
+            zeroTopAngle = 90,
+            zeroBottomAngle = 270;
+    private final double deadband = 0.001;
+    private double lastTargetVelocity = 0;
+    public SpeedMechanism2d(String key, double maxDisplayableVelocity) {
+        this.maxDisplayableVelocity = maxDisplayableVelocity;
+        this.key = key;
+        this.mechanism = new Mechanism2d(2 * maxDisplayableVelocity, 2 * maxDisplayableVelocity);
+        this.root = mechanism.getRoot(key, maxDisplayableVelocity, maxDisplayableVelocity);
+    }
+
+    public void setVelocity(double velocity) {
+        setVelocity(velocity, lastTargetVelocity);
+    }
+
+    public void setVelocity(double velocity, double targetVelocity) {
+        createLigaments(velocity, targetVelocity);
+        setTargetVelocity(targetVelocity);
+        setTargetVelocityArrowAngle(targetVelocity);
+        targetVelocityLigament.setLength(targetVelocity);
+        setCurrentVelocityArrowAngle(velocity);
+        currentVelocityLigament.setLength(velocity);
+        setLigamentColor(velocity);
+        Logger.recordOutput(key, mechanism);
+    }
+
+    public void setTargetVelocity(double targetVelocity) {
+        lastTargetVelocity = targetVelocity;
+    }
+
+    private void createLigaments(double currentVelocity, double targetVelocity) {
+        String currentVelocityLigamentKeyZ = "";
+        String targetVelocityLigamentKeyZ = "";
+        if (isCurrentVelocityLigamentBiggerThanTargetVelocityLigament(currentVelocity, targetVelocity))
+            targetVelocityLigamentKeyZ = "Z";
+        else
+            currentVelocityLigamentKeyZ = "Z";
+        this.currentVelocityLigament = root.append(new MechanismLigament2d(currentVelocityLigamentKeyZ + "CurrentVelocityLigament", 0, 0, 5, new Color8Bit(Color.kBlue)));
+        this.currentVelocityTopArrowLigament = currentVelocityLigament.append(new MechanismLigament2d(currentVelocityLigamentKeyZ + "CurrentVelocityTopArrowLigament", 0.2 * maxDisplayableVelocity, zeroTopAngle, 5, new Color8Bit(Color.kBlue)));
+        this.currentVelocityBottomArrowLigament = currentVelocityLigament.append(new MechanismLigament2d(currentVelocityLigamentKeyZ + "CurrentVelocityBottomArrowLigament", 0.2 * maxDisplayableVelocity, zeroBottomAngle, 5, new Color8Bit(Color.kBlue)));
+
+        this.targetVelocityLigament = root.append(new MechanismLigament2d(targetVelocityLigamentKeyZ + "TargetVelocityLigament", 0, 0, 5, new Color8Bit(Color.kGray)));
+        this.targetVelocityTopArrowLigament = targetVelocityLigament.append(new MechanismLigament2d(targetVelocityLigamentKeyZ + "TargetVelocityTopArrowLigament", 0.2 * maxDisplayableVelocity, zeroTopAngle, 5, new Color8Bit(Color.kGray)));
+        this.targetVelocityBottomArrowLigament = targetVelocityLigament.append(new MechanismLigament2d(targetVelocityLigamentKeyZ + "TargetVelocityBottomArrowLigament", 0.2 * maxDisplayableVelocity, zeroBottomAngle, 5, new Color8Bit(Color.kGray)));
+    }
```

Comment:
Since this is too resource inefficient (creating the new ligaments every cycle) drop this idea and just have the current be above the target at all times.